app/config/AppConfig.scala

```diff
@@ -16,25 +16,33 @@
 
 package config
 
-import config.AppConfig.getConfInt
-import play.api.Configuration
-import play.api.Play.{configuration, current}
+import play.api.Play
 import uk.gov.hmrc.play.config.ServicesConfig
 
 trait AppConfig {
-
-  protected val configuration: Configuration
-
-  private def loadConfig(key: String) = configuration.getString(key).getOrElse(throw new Exception(s"Missing configuration key: $key"))
-
-  lazy val assetsPrefix = loadConfig("assets.url") + loadConfig("assets.version")
-  lazy val analyticsToken = loadConfig("google-analytics.token")
-  lazy val analyticsHost = loadConfig("google-analytics.host")
-  lazy val devHubBaseUrl = loadConfig("devHubBaseUrl")
-  lazy val isExternalTestEnvironment = configuration.getBoolean("isExternalTestEnvironment").getOrElse(false)
-  lazy val title = if (isExternalTestEnvironment) "HMRC API Gatekeeper - Developer Sandbox" else "HMRC API Gatekeeper"
+  val assetsPrefix: String
```

Comment:
def is more flexible than val and more idiomatic - except when there is an actual value being assigned